src/cppcache/include/geode/Struct.hpp

```diff
@@ -138,7 +138,7 @@ class CPPCACHE_EXPORT Struct : public Serializable {
    * Returns the name of the field corresponding to the index number in the
    * Struct
    */
-  virtual const char* getFieldName(const int32_t index) const;
+  virtual const std::string* getFieldName(const int32_t index) const;
```

Comment:
I am too not a huge fan of returning `std::string*` but the current behavior is to return `nullptr` if the field does not exist at that index. 

@dgkimura would you be happier if this method threw [`out_of_range`](http://en.cppreference.com/w/cpp/error/out_of_range)? I think I might like that better than null.